out/notebooks/splunk.ts

```diff
@@ -56,7 +56,65 @@ export function createSearchJob(jobs, query, options) {
     });
 }
 
-export function dispatchSpl2Module(service: any, spl2Module: string, earliest: string, latest: string) {
+/**
+ * Update a module by calling the PUT /services/spl2/modules/<namespace>.<moduleName>
+ * @param service Instance of the Javascript SDK Service
+ * @param moduleName Name of the module to append to the namespace to form the request path
+ * @param namespace Full namespace to be used directly to form the request path
+ * @param module Full contents of the module to update with
+ * @returns Promise<void> (or throw Error containing data.messages[])
+ */
+export function updateSpl2Module(service: any, moduleName: string, namespace: string, module: string) {
+    // The Splunk SDK for Javascript doesn't currently support the spl2/modules endpoints
+    // nor does it support sending requests in JSON format (only receiving responses), so
+    // for now use the underlying needle library that the SDK uses for requests/responses
+    return needle(
+        'PUT',
+        // example: https://myhost.splunkcloud.com:8089/services/spl2/modules/apps.search._default
+        `${service.prefix}/services/spl2/modules/${encodeURIComponent(namespace)}.${encodeURIComponent(moduleName)}`,
```

Comment:
This is weird and Redundancy! Request parameters are also part of the payload.

Reply:
I agree 😬 - I think I have some API feedback